### ISSUE TITLE
Issue compiling xml2

### ISSUE THREAD

**Issue by fmannhardt** [r-hub/rhub#273]
I am trying to check my package with:
`rhub::check_with_sanitizers()`

However, I don't get there as the `xml2` package cannot be compiled.
I guess this is because mismatching compiler versions or something like this.

```
#> * installing *source* package ‘xml2’ ...

3180#> ** package ‘xml2’ successfully unpacked and MD5 sums checked

3181#> Found pkg-config cflags and libs!

3182#> Using PKG_CFLAGS=-I/usr/include/libxml2

3183#> Using PKG_LIBS=-lxml2

3184#> ** libs

3185#> g++ -fsanitize=undefined,bounds-strict -fno-omit-frame-pointer -std=gnu++98 -I"/usr/local/lib/R/include" -DNDEBUG -I../inst/include -I/usr/include/libxml2 -I"/home/docker/R/Rcpp/include" -I/usr/local/include -fvisibility=hidden -fpic -g -O2 -Wall -pedantic -mtune=native -c RcppExports.cpp -o RcppExports.o

3186#> In file included from /usr/include/unicode/utypes.h:38:0,

3187#> from /usr/include/unicode/ucnv_err.h:88,

3188#> from /usr/include/unicode/ucnv.h:52,

3189#> from /usr/include/libxml2/libxml/encoding.h:31,

3190#> from /usr/include/libxml2/libxml/parser.h:810,

3191#> from /usr/include/libxml2/libxml/globals.h:18,

3192#> from /usr/include/libxml2/libxml/threads.h:35,

3193#> from /usr/include/libxml2/libxml/xmlmemory.h:218,

3194#> from /usr/include/libxml2/libxml/tree.h:1307,

3195#> from ../inst/include/xml2_types.h:4,

3196#> from RcppExports.cpp:4:

3197#> /usr/include/unicode/umachine.h:347:13: error: ‘char16_t’ does not name a type; did you mean ‘wchar_t’?

3198#> typedef char16_t UChar;

3199#> ^~~~~~~~

3200#> wchar_t

3201#> In file included from /usr/include/unicode/utypes.h:39:0,

3202#> from /usr/include/unicode/ucnv_err.h:88,

3203#> from /usr/include/unicode/ucnv.h:52,

3204#> from /usr/include/libxml2/libxml/encoding.h:31,

3205#> from /usr/include/libxml2/libxml/parser.h:810,

3206#> from /usr/include/libxml2/libxml/globals.h:18,

3207#> from /usr/include/libxml2/libxml/threads.h:35,

3208#> from /usr/include/libxml2/libxml/xmlmemory.h:218,

3209#> from /usr/include/libxml2/libxml/tree.h:1307,

3210#> from ../inst/include/xml2_types.h:4,

3211#> from RcppExports.cpp:4:

3212#> /usr/include/unicode/uversion.h:167:55: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3213#> u_versionFromUString(UVersionInfo versionArray, const UChar *versionString);

3214#> ^~~~~

3215#> UChar32

3216#> In file included from /usr/include/unicode/ucnv.h:52:0,

3217#> from /usr/include/libxml2/libxml/encoding.h:31,

3218#> from /usr/include/libxml2/libxml/parser.h:810,

3219#> from /usr/include/libxml2/libxml/globals.h:18,

3220#> from /usr/include/libxml2/libxml/threads.h:35,

3221#> from /usr/include/libxml2/libxml/xmlmemory.h:218,

3222#> from /usr/include/libxml2/libxml/tree.h:1307,

3223#> from ../inst/include/xml2_types.h:4,

3224#> from RcppExports.cpp:4:

3225#> /usr/include/unicode/ucnv_err.h:198:11: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3226#> const UChar *source; /**< Pointer to the source source buffer. @stable ICU 2.0 */

3227#> ^~~~~

3228#> UChar32

3229#> /usr/include/unicode/ucnv_err.h:199:11: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3230#> const UChar *sourceLimit; /**< Pointer to the limit (end + 1) of source buffer. @stable ICU 2.0 */

3231#> ^~~~~

3232#> UChar32

3233#> /usr/include/unicode/ucnv_err.h:216:5: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3234#> UChar *target; /**< Pointer to the target buffer. @stable ICU 2.0 */

3235#> ^~~~~

3236#> UChar32

3237#> /usr/include/unicode/ucnv_err.h:217:11: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3238#> const UChar *targetLimit; /**< Pointer to the limit (end + 1) of target buffer. @stable ICU 2.0 */

3239#> ^~~~~

3240#> UChar32

3241#> /usr/include/unicode/ucnv_err.h:239:25: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3242#> const UChar* codeUnits,

3243#> ^~~~~

3244#> UChar32

3245#> /usr/include/unicode/ucnv_err.h:290:25: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3246#> const UChar* codeUnits,

3247#> ^~~~~

3248#> UChar32

3249#> /usr/include/unicode/ucnv_err.h:320:25: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3250#> const UChar* codeUnits,

3251#> ^~~~~

3252#> UChar32

3253#> /usr/include/unicode/ucnv_err.h:376:25: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3254#> const UChar* codeUnits,

3255#> ^~~~~

3256#> UChar32

3257#> In file included from /usr/include/unicode/ucnv.h:53:0,

3258#> from /usr/include/libxml2/libxml/encoding.h:31,

3259#> from /usr/include/libxml2/libxml/parser.h:810,

3260#> from /usr/include/libxml2/libxml/globals.h:18,

3261#> from /usr/include/libxml2/libxml/threads.h:35,

3262#> from /usr/include/libxml2/libxml/xmlmemory.h:218,

3263#> from /usr/include/libxml2/libxml/tree.h:1307,

3264#> from ../inst/include/xml2_types.h:4,

3265#> from RcppExports.cpp:4:

3266#> /usr/include/unicode/uenum.h:112:16: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3267#> U_STABLE const UChar* U_EXPORT2

3268#> ^~~~~

3269#> UChar32

3270#> In file included from /usr/include/unicode/ucnv.h:53:0,

3271#> from /usr/include/libxml2/libxml/encoding.h:31,

3272#> from /usr/include/libxml2/libxml/parser.h:810,

3273#> from /usr/include/libxml2/libxml/globals.h:18,

3274#> from /usr/include/libxml2/libxml/threads.h:35,

3275#> from /usr/include/libxml2/libxml/xmlmemory.h:218,

3276#> from /usr/include/libxml2/libxml/tree.h:1307,

3277#> from ../inst/include/xml2_types.h:4,

3278#> from RcppExports.cpp:4:

3279#> /usr/include/unicode/uenum.h:190:41: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3280#> uenum_openUCharStringsEnumeration(const UChar* const strings[], int32_t count,

3281#> ^~~~~

3282#> UChar32

3283#> In file included from /usr/include/libxml2/libxml/encoding.h:31:0,

3284#> from /usr/include/libxml2/libxml/parser.h:810,

3285#> from /usr/include/libxml2/libxml/globals.h:18,

3286#> from /usr/include/libxml2/libxml/threads.h:35,

3287#> from /usr/include/libxml2/libxml/xmlmemory.h:218,

3288#> from /usr/include/libxml2/libxml/tree.h:1307,

3289#> from ../inst/include/xml2_types.h:4,

3290#> from RcppExports.cpp:4:

3291#> /usr/include/unicode/ucnv.h:226:27: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3292#> const UChar* codeUnits,

3293#> ^~~~~

3294#> UChar32

3295#> /usr/include/unicode/ucnv.h:397:18: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3296#> ucnv_openU(const UChar *name,

3297#> ^~~~~

3298#> UChar32

3299#> /usr/include/unicode/ucnv.h:669:27: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3300#> const UChar *s,

3301#> ^~~~~

3302#> UChar32

3303#> /usr/include/unicode/ucnv.h:707:23: error: ‘UChar’ has not been declared

3304#> UChar *errUChars,

3305#> ^~~~~

3306#> /usr/include/unicode/ucnv.h:846:21: error: ‘UChar’ has not been declared

3307#> UChar *displayName,

3308#> ^~~~~

3309#> /usr/include/unicode/ucnv.h:1147:25: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3310#> const UChar ** source,

3311#> ^~~~~

3312#> UChar32

3313#> /usr/include/unicode/ucnv.h:1148:25: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3314#> const UChar * sourceLimit,

3315#> ^~~~~

3316#> UChar32

3317#> /usr/include/unicode/ucnv.h:1214:16: error: ‘UChar’ has not been declared

3318#> UChar **target,

3319#> ^~~~~

3320#> /usr/include/unicode/ucnv.h:1215:22: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3321#> const UChar *targetLimit,

3322#> ^~~~~

3323#> UChar32

3324#> /usr/include/unicode/ucnv.h:1252:23: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3325#> const UChar *src, int32_t srcLength,

3326#> ^~~~~

3327#> UChar32

3328#> /usr/include/unicode/ucnv.h:1283:15: error: ‘UChar’ has not been declared

3329#> UChar *dest, int32_t destCapacity,

3330#> ^~~~~

3331#> /usr/include/unicode/ucnv.h:1505:16: error: ‘UChar’ has not been declared

3332#> UChar *pivotStart, UChar **pivotSource,

3333#> ^~~~~

3334#> /usr/include/unicode/ucnv.h:1505:35: error: ‘UChar’ has not been declared

3335#> UChar *pivotStart, UChar **pivotSource,

3336#> ^~~~~

3337#> /usr/include/unicode/ucnv.h:1506:16: error: ‘UChar’ has not been declared

3338#> UChar **pivotTarget, const UChar *pivotLimit,

3339#> ^~~~~

3340#> /usr/include/unicode/ucnv.h:1506:43: error: ‘UChar’ does not name a type; did you mean ‘UChar32’?

3341#> UChar **pivotTarget, const UChar *pivotLimit,

3342#> ^~~~~

3343#> UChar32

3344#> /usr/include/unicode/ucnv.h:1916:46: error: ‘UChar’ has not been declared

3345#> ucnv_fixFileSeparator(const UConverter *cnv, UChar *source, int32_t sourceLen);

3346#> ^~~~~

3347#> make: *** [/usr/local/lib/R/etc/Makeconf:168: RcppExports.o] Error 1

3348#> ERROR: compilation failed for package ‘xml2’
```

**Comment by gaborcsardi** [comment]
I think this is probably `https://github.com/r-lib/xml2/issues/231`

Any chance that we can fix it, @jimhester ?

**Comment by jimhester** [comment]
I am not sure how we could fix it....

**Comment by gaborcsardi** [comment]
Isn't it enough to add C++-11 to system requirements? 

c.f. https://cran.r-project.org/doc/manuals/R-exts.html#Using-C_002b_002b11-code

**Comment by jimhester** [comment]
That seems a pretty onerous system requirement

**Comment by gaborcsardi** [comment]
Got it. So this is an Ubuntu bug then. Can we change the compiler flag only for the problematic platform(s)? xml2 already has a configure script, so maybe we could do it there? 

I understand that this is a pain, but maybe it does not take too much to fix it, Ubuntu 16.04 will be with use for some time....

**Comment by jimhester** [comment]
We could, but I haven't looked into it enough to see if there is a better way forward than just using C++11.

**Comment by jackwasey** [comment]
The following seems to work. I added -Wno-ignored-attributes just to reduce the massive Eigen compilation noise. 

```
Rscript -e 'rhub::check_with_sanitizers(env_vars = c(MAKEFLAGS = "CXX11FLAGS+=-std=c++11\\ -Wno-ignored-attributes CXXFLAGS+=-std=c++11\\ -Wno-ignored-attributes"))'
```

**Comment by gaborcsardi** [comment]
@jackwasey good stuff, thanks!

**Comment by Danish00786** [comment]
I got the same error during installing root_v5.34.36 on ubuntu 18.04.

Scanning dependencies of target MathCore
[ 32%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/G__MathFit.cxx.o
[ 32%] Building CXX object core/thread/CMakeFiles/Thread.dir/src/TPosixThread.cxx.o
[ 32%] Linking CXX shared library ../../lib/libReflexDict.so
[ 32%] Building CXX object core/thread/CMakeFiles/Thread.dir/src/TPosixThreadFactory.cxx.o
[ 32%] Built target ReflexDict
[ 32%] Generating G__Smatrix.cxx, G__Smatrix.h
[ 32%] Linking CXX shared library ../../lib/libThread.so
[ 32%] Built target Thread
[ 32%] Generating G__GenVector.cxx, G__GenVector.h
[ 32%] Linking CXX shared library ../../lib/libCintex.so
[ 33%] Generating G__Smatrix32.cxx, G__Smatrix32.h
[ 33%] Built target Cintex
[ 33%] Generating G__FFTW.cxx, G__FFTW.h
Scanning dependencies of target FFTW
[ 33%] Building CXX object math/fftw/CMakeFiles/FFTW.dir/G__FFTW.cxx.o
Scanning dependencies of target Smatrix
[ 33%] Building CXX object math/smatrix/CMakeFiles/Smatrix.dir/G__Smatrix32.cxx.o
[ 33%] Building CXX object math/fftw/CMakeFiles/FFTW.dir/src/TFFTComplex.cxx.o
[ 33%] Building CXX object math/fftw/CMakeFiles/FFTW.dir/src/TFFTComplexReal.cxx.o
[ 33%] Generating G__GenVector32.cxx, G__GenVector32.h
[ 33%] Building CXX object math/fftw/CMakeFiles/FFTW.dir/src/TFFTReal.cxx.o
[ 33%] Building CXX object math/fftw/CMakeFiles/FFTW.dir/src/TFFTRealComplex.cxx.o
[ 33%] Linking CXX shared library ../../lib/libFFTW.so
[ 33%] Built target FFTW
[ 33%] Generating G__XMLParser.cxx, G__XMLParser.h
Scanning dependencies of target XMLParser
[ 33%] Building CXX object io/xmlparser/CMakeFiles/XMLParser.dir/G__XMLParser.cxx.o
[ 33%] Building CXX object io/xmlparser/CMakeFiles/XMLParser.dir/src/TDOMParser.cxx.o
Scanning dependencies of target GenVector
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/G__GenVector32.cxx.o
In file included from /usr/include/unicode/utypes.h:38:0,
                 from /usr/include/unicode/ucnv_err.h:88,
                 from /usr/include/unicode/ucnv.h:52,
                 from /usr/include/libxml2/libxml/encoding.h:31,
                 from /usr/include/libxml2/libxml/parser.h:810,
                 from /usr/include/libxml2/libxml/globals.h:18,
                 from /usr/include/libxml2/libxml/threads.h:35,
                 from /usr/include/libxml2/libxml/xmlmemory.h:218,
                 from /usr/include/libxml2/libxml/tree.h:1307,
                 from /home/danisih/root/io/xmlparser/src/TDOMParser.cxx:29:
/usr/include/unicode/umachine.h:347:13: error: ‘char16_t’ does not name a type
     typedef char16_t UChar;
             ^
In file included from /usr/include/unicode/utypes.h:39:0,
                 from /usr/include/unicode/ucnv_err.h:88,
                 from /usr/include/unicode/ucnv.h:52,
                 from /usr/include/libxml2/libxml/encoding.h:31,
                 from /usr/include/libxml2/libxml/parser.h:810,
                 from /usr/include/libxml2/libxml/globals.h:18,
                 from /usr/include/libxml2/libxml/threads.h:35,
                 from /usr/include/libxml2/libxml/xmlmemory.h:218,
                 from /usr/include/libxml2/libxml/tree.h:1307,
                 from /home/danisih/root/io/xmlparser/src/TDOMParser.cxx:29:
/usr/include/unicode/uversion.h:167:55: error: ‘UChar’ does not name a type
 u_versionFromUString(UVersionInfo versionArray, const UChar *versionString);
                                                       ^
In file included from /usr/include/unicode/ucnv.h:52:0,
                 from /usr/include/libxml2/libxml/encoding.h:31,
                 from /usr/include/libxml2/libxml/parser.h:810,
                 from /usr/include/libxml2/libxml/globals.h:18,
                 from /usr/include/libxml2/libxml/threads.h:35,
                 from /usr/include/libxml2/libxml/xmlmemory.h:218,
                 from /usr/include/libxml2/libxml/tree.h:1307,
                 from /home/danisih/root/io/xmlparser/src/TDOMParser.cxx:29:
/usr/include/unicode/ucnv_err.h:198:11: error: ‘UChar’ does not name a type
     const UChar *source;        /**< Pointer to the source source buffer. @stable ICU 2.0    */
           ^
/usr/include/unicode/ucnv_err.h:199:11: error: ‘UChar’ does not name a type
     const UChar *sourceLimit;   /**< Pointer to the limit (end + 1) of source buffer. @stable ICU 2.0    */
           ^
/usr/include/unicode/ucnv_err.h:216:5: error: ‘UChar’ does not name a type
     UChar *target;              /**< Pointer to the target buffer. @stable ICU 2.0    */
     ^
/usr/include/unicode/ucnv_err.h:217:11: error: ‘UChar’ does not name a type
     const UChar *targetLimit;   /**< Pointer to the limit (end + 1) of target buffer. @stable ICU 2.0     */
           ^
/usr/include/unicode/ucnv_err.h:239:25: error: ‘UChar’ does not name a type
                   const UChar* codeUnits,
                         ^
/usr/include/unicode/ucnv_err.h:290:25: error: ‘UChar’ does not name a type
                   const UChar* codeUnits,
                         ^
/usr/include/unicode/ucnv_err.h:320:25: error: ‘UChar’ does not name a type
                   const UChar* codeUnits,
                         ^
/usr/include/unicode/ucnv_err.h:376:25: error: ‘UChar’ does not name a type
                   const UChar* codeUnits,
                         ^
In file included from /usr/include/unicode/ucnv.h:53:0,
                 from /usr/include/libxml2/libxml/encoding.h:31,
                 from /usr/include/libxml2/libxml/parser.h:810,
                 from /usr/include/libxml2/libxml/globals.h:18,
                 from /usr/include/libxml2/libxml/threads.h:35,
                 from /usr/include/libxml2/libxml/xmlmemory.h:218,
                 from /usr/include/libxml2/libxml/tree.h:1307,
                 from /home/danisih/root/io/xmlparser/src/TDOMParser.cxx:29:
/usr/include/unicode/uenum.h:112:16: error: ‘UChar’ does not name a type
 U_STABLE const UChar* U_EXPORT2
                ^
In file included from /usr/include/unicode/ucnv.h:53:0,
                 from /usr/include/libxml2/libxml/encoding.h:31,
                 from /usr/include/libxml2/libxml/parser.h:810,
                 from /usr/include/libxml2/libxml/globals.h:18,
                 from /usr/include/libxml2/libxml/threads.h:35,
                 from /usr/include/libxml2/libxml/xmlmemory.h:218,
                 from /usr/include/libxml2/libxml/tree.h:1307,
                 from /home/danisih/root/io/xmlparser/src/TDOMParser.cxx:29:
/usr/include/unicode/uenum.h:190:41: error: ‘UChar’ does not name a type
 uenum_openUCharStringsEnumeration(const UChar* const strings[], int32_t count,
                                         ^
In file included from /usr/include/libxml2/libxml/encoding.h:31:0,
                 from /usr/include/libxml2/libxml/parser.h:810,
                 from /usr/include/libxml2/libxml/globals.h:18,
                 from /usr/include/libxml2/libxml/threads.h:35,
                 from /usr/include/libxml2/libxml/xmlmemory.h:218,
                 from /usr/include/libxml2/libxml/tree.h:1307,
                 from /home/danisih/root/io/xmlparser/src/TDOMParser.cxx:29:
/usr/include/unicode/ucnv.h:226:27: error: ‘UChar’ does not name a type
                     const UChar* codeUnits,
                           ^
/usr/include/unicode/ucnv.h:397:18: error: ‘UChar’ does not name a type
 ucnv_openU(const UChar *name,
                  ^
/usr/include/unicode/ucnv.h:669:27: error: ‘UChar’ does not name a type
                     const UChar *s,
                           ^
/usr/include/unicode/ucnv.h:707:23: error: ‘UChar’ has not been declared
                       UChar *errUChars,
                       ^
/usr/include/unicode/ucnv.h:846:21: error: ‘UChar’ has not been declared
                     UChar *displayName,
                     ^
/usr/include/unicode/ucnv.h:1147:25: error: ‘UChar’ does not name a type
                   const UChar ** source,
                         ^
/usr/include/unicode/ucnv.h:1148:25: error: ‘UChar’ does not name a type
                   const UChar * sourceLimit,
                         ^
/usr/include/unicode/ucnv.h:1214:16: error: ‘UChar’ has not been declared
                UChar **target,
                ^
/usr/include/unicode/ucnv.h:1215:22: error: ‘UChar’ does not name a type
                const UChar *targetLimit,
                      ^
/usr/include/unicode/ucnv.h:1252:23: error: ‘UChar’ does not name a type
                 const UChar *src, int32_t srcLength,
                       ^
/usr/include/unicode/ucnv.h:1283:15: error: ‘UChar’ has not been declared
               UChar *dest, int32_t destCapacity,
               ^
/usr/include/unicode/ucnv.h:1505:16: error: ‘UChar’ has not been declared
                UChar *pivotStart, UChar **pivotSource,
                ^
/usr/include/unicode/ucnv.h:1505:35: error: ‘UChar’ has not been declared
                UChar *pivotStart, UChar **pivotSource,
                                   ^
/usr/include/unicode/ucnv.h:1506:16: error: ‘UChar’ has not been declared
                UChar **pivotTarget, const UChar *pivotLimit,
                ^
/usr/include/unicode/ucnv.h:1506:43: error: ‘UChar’ does not name a type
                UChar **pivotTarget, const UChar *pivotLimit,
                                           ^
/usr/include/unicode/ucnv.h:1916:46: error: ‘UChar’ has not been declared
 ucnv_fixFileSeparator(const UConverter *cnv, UChar *source, int32_t sourceLen);
                                              ^
io/xmlparser/CMakeFiles/XMLParser.dir/build.make:100: recipe for target 'io/xmlparser/CMakeFiles/XMLParser.dir/src/TDOMParser.cxx.o' failed
make[2]: *** [io/xmlparser/CMakeFiles/XMLParser.dir/src/TDOMParser.cxx.o] Error 1
CMakeFiles/Makefile2:6446: recipe for target 'io/xmlparser/CMakeFiles/XMLParser.dir/all' failed
make[1]: *** [io/xmlparser/CMakeFiles/XMLParser.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/G__GenVector.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/G__MathCore.cxx.o
[ 34%] Building CXX object math/smatrix/CMakeFiles/Smatrix.dir/G__Smatrix.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/3DConversions.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/3DDistances.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/AxisAngle.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/AxisAngleXother.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/BitReproducible.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/Boost.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/BoostX.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/BoostY.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/BoostZ.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/EulerAngles.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/GenVector_exception.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/LorentzRotation.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/Plane3D.cxx.o
[ 34%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/Quaternion.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/G__Math.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/AdaptiveIntegratorMultiDim.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/BasicMinimizer.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/BinData.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/BinPoint.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/BrentMethods.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/BrentMinimizer1D.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/BrentRootFinder.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/ChebyshevPol.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/DataRange.cxx.o
[ 34%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/DataVector.cxx.o
[ 35%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/DistSampler.cxx.o
[ 35%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/DistSamplerOptions.cxx.o
[ 35%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/Factory.cxx.o
[ 35%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/FitConfig.cxx.o
[ 35%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/FitResult.cxx.o
[ 35%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/FitUtil.cxx.o
[ 35%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/FitUtilParallel.cxx.o
[ 35%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/QuaternionXaxial.cxx.o
[ 35%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/Rotation3D.cxx.o
[ 35%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/Rotation3DxAxial.cxx.o
[ 35%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/RotationZYX.cxx.o
[ 35%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/Fitter.cxx.o
[ 35%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/Transform3D.cxx.o
[ 35%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/Translation3D.cxx.o
[ 36%] Building CXX object math/genvector/CMakeFiles/GenVector.dir/src/VectorUtil.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/FunctorCint.cxx.o
[ 36%] Linking CXX shared library ../../lib/libGenVector.so
[ 36%] Built target GenVector
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/GaussIntegrator.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/GaussLegendreIntegrator.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/GenAlgoOptions.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/GoFTest.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/Integrator.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/IntegratorOptions.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MinimTransformFunction.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MinimizerOptions.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/MinimizerVariableTransformation.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/ParameterSettings.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/PdfFuncMathCore.cxx.o
[ 36%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/ProbFuncMathCore.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/QuantFuncMathCore.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/RichardsonDerivator.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/RootFinder.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/SparseData.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/SpecFuncCephes.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/SpecFuncCephesInv.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/SpecFuncMathCore.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TComplex.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TDataPointN.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TKDTree.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TKDTreeBinning.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TMath.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandom.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandom1.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandom2.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TRandom3.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TStatistic.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/TVirtualFitter.cxx.o
[ 37%] Building CXX object math/mathcore/CMakeFiles/MathCore.dir/src/UnBinData.cxx.o
[ 37%] Linking CXX shared library ../../lib/libMathCore.so
[ 37%] Built target MathCore
[ 37%] Linking CXX shared library ../../lib/libSmatrix.so
[ 37%] Built target Smatrix
Makefile:151: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by jimhester** [comment]
I fixed this in xml2 by explicitly declaring `UCHAR_TYPE` https://github.com/r-lib/xml2/commit/e051ffa6577f5fdd6e3a565981791d54fab74140, you could suggest the root maintainers do the same, or define it yourself in your make flags.

I think this general issue can be closed though, this is fixed as of xml2 1.2.1
